### DISCUSSION TITLE
custom validation message

### DISCUSSION THREAD

**Discussion by vinodhum** [react-hook-form/react-hook-form#2226] [Q&A] [answered]
Hi,
how to set custom validation message on rules with controller
 validate: {
      value: isUnique, # this returns true or false
      message: 'Nickname must be unique',
    },

above is throwing me error "**Uncaught (in promise) TypeError: validateFunction is not a function**"

but if set just
 validate: true it works but I cant set custom message 

how can I set custom message ?

**Comment by bluebill1049** [comment]
validate is a function 
<img width="966" alt="Screen Shot 2020-07-15 at 9 49 03 am" src="https://user-images.githubusercontent.com/10513364/87487214-6b81be80-c680-11ea-856c-56e994735336.png">

**Reply by vinodhum** [reply]
how can i dot it on rules on controller ? as I am not using register, secondly, I dont see option to set custom message for each validation

**Reply by vinodhum** [reply]
secondly, custom validation not triggering onblure

**Comment by Oxyps** [comment, marked as answer]
There is a way to set message to custom validation, like in required?
``` js
rules={{
  required: 'Campo necessário',
  validate: () => {
    const date = getValues('inDate');
    return date?.getTime() >= new Date('1999-01-01');
  }
}}
```

**Reply by bluebill1049** [reply]
date?.getTime() < new Date('1999-01-01') && 'something is wrong'

**Reply by Oxyps** [reply]
Uhh nice man, thank you!
